drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,826 @@
+
+    /* Restore previous operating mode */
+    sx127x_reg_write(dev, SX127X_REG_OPMODE, prev_mode);
+}
+
+uint32_t sx127x_get_time_on_air(sx127x_t *dev)
+{
+    uint32_t air_time = 0;
+
+    uint8_t pkt_len = dev->settings.time_on_air_pkt_len;
+    switch (dev->settings.modem) {
+        case SX127X_MODEM_FSK:
+            /* todo */
+            break;
+        case SX127X_MODEM_LORA:
+        {
+            double bw = 0.0;
+
+            /* Note: When using LoRa modem only bandwidths 125, 250 and 500 kHz are supported. */
+            switch (dev->settings.lora.bandwidth) {
+                case SX127X_BW_125_KHZ:
+                    bw = 125e3;
+                    break;
+                case SX127X_BW_250_KHZ:
+                    bw = 250e3;
+                    break;
+                case SX127X_BW_500_KHZ:
+                    bw = 500e3;
+                    break;
+                default:
+                    DEBUG("Invalid bandwith: %d\n", dev->settings.lora.bandwidth);
+                    break;
+            }
+
+            /* Symbol rate : time for one symbol [secs] */
+            double rs = bw / (1 << dev->settings.lora.datarate);
+            double ts = 1 / rs;
+
+            /* time of preamble */
+            double t_preamble = (dev->settings.lora.preamble_len + 4.25) * ts;
+
+            /* Symbol length of payload and time */
+            double tmp =
+                ceil(
+                    (8 * pkt_len - 4 * dev->settings.lora.datarate + 28
+                     + 16 * dev->settings.lora.crc_on
+                     - (!dev->settings.lora.use_fix_len ? 20 : 0))
+                    / (double) (4 * dev->settings.lora.datarate
+                                - ((dev->settings.lora.low_datarate_optimize
+                                    > 0) ? 2 : 0)))
+                * (dev->settings.lora.coderate + 4);
+            double n_payload = 8 + ((tmp > 0) ? tmp : 0);
+            double t_payload = n_payload * ts;
+
+            /* Time on air */
+            double t_on_air = t_preamble + t_payload;
+
+            /* return seconds */
+            air_time = floor(t_on_air * 1e3 + 0.999);
+        }
+        break;
+    }
+
+    return air_time;
+}
+
+void sx127x_set_sleep(sx127x_t *dev)
+{
+    DEBUG("[DEBUG] Set sleep\n");
+
+    /* Disable running timers */
+    xtimer_remove(&dev->_internal.tx_timeout_timer);
+    xtimer_remove(&dev->_internal.rx_timeout_timer);
+    xtimer_remove(&dev->_internal.rx_timeout_syncword_timer);
+
+    /* Put chip into sleep */
+    sx127x_set_op_mode(dev, SX127X_RF_OPMODE_SLEEP);
+    sx127x_set_state(dev,  SX127X_RF_IDLE);
+}
+
+void sx127x_set_standby(sx127x_t *dev)
+{
+    DEBUG("[DEBUG] Set standby\n");
+
+    /* Disable running timers */
+    xtimer_remove(&dev->_internal.tx_timeout_timer);
+    xtimer_remove(&dev->_internal.rx_timeout_timer);
+    xtimer_remove(&dev->_internal.rx_timeout_syncword_timer);
```

Comment:
Something I forgot to remove when I removed the FSK part : I started to look into this but realized I won't be able to finish it for the release.